book/src/dev/rfcs/0006-contextual-difficulty.md

```diff
@@ -119,7 +120,13 @@ TODO:
   - check how zcashd implements signed median time differences
   - open a ticket to update the Zcash spec
 
-## The relevant chain
+## State service interface changes
+[state-service-interface]: #state-service-interface
+
+Contextual validation accesses recent blocks. So we modify the the internal state
+service interface to provide an abstraction for accessing recent blocks.
```

Comment:
I was gonna say it already has this abstraction but then I realized that it doesn't, because the current one only works with best chains

https://github.com/teor2345/zebra/blob/contextual-difficulty-rfc/zebra-state/src/service.rs#L230-L236

Reply:
Thinking about it I think we might want to add an iterator API to the two state types, then we can write the following

```rust
let blocks = self.mem.chain_iter(&block.header.previous_block_hash).chain(self.sled.chain_iter()).take(28);
// ...
```

Reply:
Resolved by the implementation in PR #1271